### DISCUSSION TITLE
Any way to draw particles of different radii fast?

### DISCUSSION THREAD

**Discussion by Dadmehr-Ghasemfar** [taichi-dev/taichi#8586] [Docs]
I am trying to draw spheres of different radii. When I use the particles function below, they can only be set to ONE radius.

![Screenshot 2024-08-29 233655](https://github.com/user-attachments/assets/ba739142-d89a-44a9-93a8-f53f49c88958)

I fixed this by manually looping through the particles and adding each to the scene individually, as shown below. This ends up being significantly slower runtime.

![Screenshot 2024-08-29 233738](https://github.com/user-attachments/assets/eb308b73-f265-44e6-96b2-366dda32d79f)

Ideally, I want something like this, but this does not seem to be possible.

![image](https://github.com/user-attachments/assets/f04b2749-aeae-4ca5-92c0-e838f7a8c943)

**How can I draw many particles of different sizes without sacrificing runtime by manually looping??**

For context, here is how I define the above fields

![Screenshot 2024-08-29 235904](https://github.com/user-attachments/assets/7c071c26-caab-4344-9ee0-7e0f16eb691f)

**Comment by jarmitage** [comment]
You might want to check out the drawing module (`tv.px`) in [Tölvera](https://github.com/afhverjuekki/tolvera/):
- https://afhverjuekki.github.io/tolvera/reference/tolvera/pixels/#tolvera.pixels.Pixels.circles
